### PR TITLE
Scan interior pointers from GC stack roots when hosted by a native GC host

### PR DESCRIPTION
**Pull request by sanketj**
Currently, we skip scanning interior pointers from GC stack roots if we are not running script (or if we are in mem protect mode). When host objects are also placed on the Chakra heap for GC purposes, we must scan interior pointers from the stack even when not running script. To scope this fix to just native GC host scenarios, and to avoid regressing JSRT scenarios, we are introducing a bit on the recycler that will be set for native GC hosts. If this bit is set, we will always scan interior pointers from the stack, and if it is not set, we will continue doing what we do currently.

In order to set this bit, an API is being exposed on the JS thread service that is published and can be called by the host.

### CODE REVIEW

File: lib/Common/Memory/Recycler.cpp [hunk 1]
```diff
@@ -937,6 +937,18 @@ Recycler::SetIsInScript(bool isInScript)
     this->isInScript = isInScript;
 }
 
+bool
+Recycler::HasNativeGCHost() const
+{
+    return this->hasNativeGCHost;
```
Comment (dlibby-) [inline lib/Common/Memory/Recycler.cpp]: should we initialize in the constructor for completeness?

**Comment by dilijev** [comment]
Marked as `External Contributor` as a reminder that a Chakra team member needs to adopt, test, and merge this change.

File: lib/Common/Memory/Recycler.cpp (hunk 1 above)
Comment (sanketj) [reply lib/Common/Memory/Recycler.cpp]: Yes.

**Comment by dilijev** [comment]
Moved to https://github.com/Microsoft/ChakraCore/pull/5403

**Pull request closed without merging**